pkg/networkservice/kernelutils/common.go

```diff
@@ -0,0 +1,89 @@
+// Copyright 2019 SUSE LLC.
+// SPDX-License-Identifier: Apache-2.0
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at:
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package kernelutils
+
+import (
+	"github.com/networkservicemesh/networkservicemesh/controlplane/api/connectioncontext"
+
+  "net"
+
+  "github.com/sirupsen/logrus"
+  "github.com/vishvananda/netlink"
+)
+
+// Kernel forwarding plane related constants
+const (
+	cLOCAL    = iota
+	cINCOMING = iota
+	cOUTGOING = iota
+)
+
+const (
+	/* VETH pairs are used only for local connections(same node), so we can use a larger MTU size as there's no multi-node connection */
+	cVETHMTU    = 16000
+	cCONNECT    = true
+	cDISCONNECT = false
+)
+
+type connectionConfig struct {
```

Comment:
Why define a new struct here rather than simply using the stuff already in the Connection struct from the api?

Reply:
I saw this laying around, really a relic of copy/paste just to get some momentum built. I have to confess I'm not familiar with the API's available to me, I'm learning this as I go. Keep pointing out places where there are structs and things that can be re-used.

Reply:
All good... one of the things I am seeking to encourage is to keep things simple :)  If this isn't buying you anything, might be a good idea to simplify it out :)